src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ConnectSqlCmdCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlServer.Management.Common;
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Data.SqlClient;
+
+namespace Microsoft.SqlTools.ServiceLayer.BatchParser
+{
+    public class ConnectSqlCmdCommand : SqlCmdCommand
+    {
+        internal ConnectSqlCmdCommand(string server, string username, string password) : base(LexerTokenType.Connect)
+        {
+            Server = server;
+            UserName = username;
+            Password = password;
+        }
+
+        public string Server { get; private set; }
+        public string UserName { get; private set; }
+        public string Password { get; private set; }
+
+
+        /// <summary>
+        /// attempts to establish connection with given params
+        /// </summary>
+        /// <returns>returns the connection object is successful, throws otherwise</returns>
+        public DbConnection Connect()
+        {
+            //create SqlConnectionInfo object
+            SqlConnectionInfo tempConnectionInfo = new SqlConnectionInfo();
```

Comment:
nit: change the variable name to connectionInfo. I do not see anything temporary about the connection info.